src/components/elements/query-image.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import { GatsbyImage, getImage } from 'gatsby-plugin-image'
+
+export const ImageWrapper = styled.div<ImageWrapperProps>`
+    & .gatsby-image-wrapper {
+        width: ${(props) => props.width || '100%'};
+        height: ${(props) => props.height};
+    }
+`
+
+type QueryImageProps = {
```

Comment:
doesn't it better to move the types to the top level under imports?
just to achieve more readability.
because you used ImageWrapperProps before defining it.